Week3/Homework/task3_morejs.js

```diff
@@ -0,0 +1,260 @@
+//    called as vehicleType("blue", 2)
+
+function vehicleType(color, code) {
+    const vehicle = code === 1 ? "car" : code === 2 ? "motorbike" : "invalid";
+    if (vehicle === "invalid") {
+        console.log("Please provide a valid code!");
+    } else {
+        console.log("a " + color + " " + vehicle);
+    }
+}
+
+vehicleType("black", 1);
+
+// 5. Can you write the following without the if statement, but with just as 
+//    a single line with console.log(...);?
+
+// if (3 === 3) {
+//     console.log("yes");
+//   } else {
+//     console.log("no");
+//   }
+
+console.log(3 === 3 ? "yes" : "no");
+
+// 6. Create a function called vehicle, like before, but takes another parameter 
+//    called age, so that vehicle("blue", 1, 5) prints 'a blue used car'
+
+function vehicle(color, code, age) {
+    const vehicle = code === 1 ? "car" : code === 2 ? "motorbike" : "invalid";
+    if (vehicle === "invalid") {
+        console.log("Please provide a valid code!");
+    } else if (typeof age !== "number" || age < 0) {
+        console.log("Please provide a valid age!");
+    } else if (age > 1) {
+        console.log("a " + color + " used " + vehicle);
+    } else {
+        console.log("a " + color + " new " + vehicle);
+    }
+}
+
+vehicle("green", 2, 1);
+
+// 7. Make a list of vehicles, you can add "motorbike", "caravan", "bike", or more.
+
+let vehicles = ["motorbike", "caravan", "bike", "car", "UFO"];
+
+// 8. How do you get the third element from that list?
+
+const thirdElement = vehicles[2];
+console.log(thirdElement);
+
+// 9. Change the function vehicle to use the list of question 7. 
+//    So that vehicle("green", 3, 1) prints "a green new bike".
+
+function specialVehicle(color, code, age) {
+    const vehicle = vehicles[code - 1];
+    if (vehicle === undefined) {
+        console.log("No vehicle for this element! The code must be between 1 and 5!");
+    } else if (typeof age !== "number" || age < 0) {
+        console.log("Please provide a valid age!");
+    } else if (age > 1) {
+        console.log("a " + color + " used " + vehicle);
+    } else {
+        console.log("a " + color + " new " + vehicle);
+    }
+}
+
+specialVehicle("yellow", 3, 2);
+
+// 10. Use the list of vehicles to write an advertisement. So that it prints something 
+//     like: "Amazing Joe's Garage, we service cars, motorbikes, caravans and bikes.". 
+//     (Hint: use a for loop.)
+
+// using join()
+function advertiseWithJoin() {
+    const vehicleList = vehicles.join("s, ");
+    console.log("Amazing Joe's Garage, we service " + vehicleList + "s.");
+}
+
+advertiseWithJoin();
+
+// using for loop
+function advertiseWithLoop() {
+    let advertisementText = "Amazing Joe's Garage, we service ";
+    for (let e = 0; e < vehicles.length; e++) {
+        if (e !== vehicles.length - 1) {
+            advertisementText += vehicles[e] + "s, ";
+        } else {
+            advertisementText += vehicles[e] + "s.";
+        }
+    }
```

Comment:
Good solution, you can also do it like this, but it's more complex so you don't have to change your code:) :

```
  for (let e = 0; e < vehicles.length; e++) {
    advertisementText += vehicles[e] + "s";
    advertisementText += (e === vehicles.length - 1) ? "." : ",";
   }
```

Reply:
nice one indeed, thanks!